test/utility/test_util.h

```diff
@@ -139,6 +139,17 @@ typedef struct {
             testutil_die(__r, "%s/%d: %s", __PRETTY_FUNCTION__, __LINE__, #call); \
     } while (0)
 
+/*
+ * testutil_check_error_ok --
+ *     Complain and quit if a function call fails, with specified error ok.
+ */
+#define testutil_check_error_ok(call, e)                                          \
+    do {                                                                          \
+        int __r;                                                                  \
+        if ((__r = (call)) != 0 && (__r = (call)) != (e))                         \
+            testutil_die(__r, "%s/%d: %s", __PRETTY_FUNCTION__, __LINE__, #call); \
+    } while (0)
```

Comment:
This repeats the function call. I think you want:
```
diff --git a/test/utility/test_util.h b/test/utility/test_util.h
index 0a049c009..c48238f09 100644
--- a/test/utility/test_util.h
+++ b/test/utility/test_util.h
@@ -146,7 +146,7 @@ typedef struct {
 #define testutil_check_error_ok(call, e)                                          \
     do {                                                                          \
         int __r;                                                                  \
-        if ((__r = (call)) != 0 && (__r = (call)) != (e))                         \
+        if ((__r = (call)) != 0 && __r != (e))                                    \
             testutil_die(__r, "%s/%d: %s", __PRETTY_FUNCTION__, __LINE__, #call); \
     } while (0)
 
```

Reply:
My bad, thanks for pointing this out.